src/main/java/net/snowflake/client/core/JsonSqlOutput.java

```diff
@@ -0,0 +1,390 @@
+          schema
+              .getFields()
+              .add(buildSchemaWithScaleAndPrecision(fieldName, "real", 0, 0, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeDouble(double value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          json.put(fieldName, value);
+          schema
+              .getFields()
+              .add(buildSchemaWithScaleAndPrecision(fieldName, "real", 0, 0, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeBigDecimal(BigDecimal value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          json.put(fieldName, value);
+          schema
+              .getFields()
+              .add(buildSchemaWithScaleAndPrecision(fieldName, "real", 0, 0, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeBytes(byte[] value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          json.put(fieldName, new SFBinary(value).toHex());
+          schema
+              .getFields()
+              .add(FieldSchemaCreator.buildSchemaForBytesType(fieldName, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeDate(Date value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          json.put(
+              fieldName,
+              ResultUtil.getDateAsString(value, getDateTimeFormat("DATE_OUTPUT_FORMAT")));
+          schema
+              .getFields()
+              .add(FieldSchemaCreator.buildSchemaTypeAndNameOnly(fieldName, "date", maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeTime(Time x) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          final long MS_IN_DAY = 86400 * 1000;
+          long msSinceEpoch = x.getTime();
+          long msSinceMidnight = (msSinceEpoch % MS_IN_DAY + MS_IN_DAY) % MS_IN_DAY;
+          long nanosSinceMidnight = msSinceMidnight * 1000 * 1000;
+          String result =
+              ResultUtil.getSFTimeAsString(
+                  SFTime.fromNanoseconds(nanosSinceMidnight),
+                  9,
+                  getDateTimeFormat("TIME_OUTPUT_FORMAT"));
+
+          json.put(fieldName, result);
+          schema
+              .getFields()
+              .add(buildSchemaWithScaleAndPrecision(fieldName, "time", 9, 0, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeTimestamp(Timestamp value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          SnowflakeType snowflakeType =
+              SnowflakeType.fromString(
+                  maybeColumn
+                      .map(cl -> cl.type())
+                      .filter(str -> !str.isEmpty())
+                      .orElse("TIMESTAMP"));
+          int columnType = SnowflakeUtil.EXTRA_TYPES_TIMESTAMP_TZ;
+          TimeZone timeZone = TimeZone.getDefault();
```

Comment:
For timestampTz shouldn't we take it from timestamp?

Reply:
Timestamp hasn't api to return timezone, true?

Reply:
I should be implemented extra method to add timezone, as a default a session timezone is set